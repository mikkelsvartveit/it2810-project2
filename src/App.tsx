import LoginPage from "./pages/LoginPage";
import StatsPage from "./pages/StatsPage";
import { Routes, Route, HashRouter } from "react-router-dom";

function App() {
  return (
    <HashRouter>
      <Routes>
        <Route path="/" element={<LoginPage />} />
        <Route path="/stats" element={<StatsPage />} />
      </Routes>
    </HashRouter>
  );
}

export default App;
