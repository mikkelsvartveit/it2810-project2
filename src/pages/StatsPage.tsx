import { useContext, useState, useEffect } from "react";
import { useNavigate } from "react-router-dom";
import { Tab, TabList, TabPanel, Tabs } from "react-tabs";
import { RepoContext } from "../App";
import GraphsComp from "../components/GraphsComp";
import { Leaderboard } from "../components/Leaderboard";
import Navbar from "../components/Navbar";
import "../styles/react-tabs.scss";

const InfoPage = () => {
  const [isLoading, setLoading] = useState(true);
  const repoContext = useContext(RepoContext);
  const navigate = useNavigate();

  useEffect(() => {
    if (!repoContext.repoData.repoURI || !repoContext.repoData.repoToken) {
      navigate("/");
    }

    setLoading(false);
  }, [navigate, repoContext]);

  const handleLogout = () => {
    repoContext.setRepoData({ repoURI: null, repoToken: null });
    navigate("/");
  };

  if (isLoading) {
    return <div className="App">Loading...</div>;
  } else {
    return (
      <>
        <Navbar />

        <div className="content">
          <Tabs>
            <TabList>
              <Tab>Leaderboard</Tab>
              <Tab>Graphs</Tab>
            </TabList>

            <TabPanel>
              <Leaderboard />
            </TabPanel>

            <TabPanel>
              <h2>Graphs</h2>
              <GraphsComp />
            </TabPanel>
          </Tabs>
        </div>
      </>
    );
  }
};
export default InfoPage;
