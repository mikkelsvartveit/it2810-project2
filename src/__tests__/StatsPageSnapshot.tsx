import { BrowserRouter } from "react-router-dom";
import StatsPage from "../pages/StatsPage";
import {
  screen,
  render,
  waitForElementToBeRemoved,
} from "@testing-library/react";
import { RepoContext } from "../App";

it("Stats page renders correctly", async () => {
  const { asFragment } = render(
    <BrowserRouter>
      <RepoContext.Provider
        value={{
          repoData: {
            repoURI: "hei",
            repoToken: "test",
          },
          setRepoData: () => {},
        }}
      >
        <StatsPage />
      </RepoContext.Provider>
    </BrowserRouter>
  );
  await waitForElementToBeRemoved(screen.queryByText("Loading..."));
  expect(asFragment()).toMatchSnapshot();
});
