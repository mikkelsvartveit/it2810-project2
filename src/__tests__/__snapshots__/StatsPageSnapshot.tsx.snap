// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats page renders correctly 1`] = `
<DocumentFragment>
  <div
    class="navbar"
  >
    <div
      class="navbar-content"
    >
      <div>
        <span
          class="navbar-title"
        >
          GitLab Analyzer
        </span>
      </div>
      <button>
        <a
          href="/"
        />
        Exit repo
      </button>
    </div>
  </div>
  <div
    class="content"
  >
    <div
      class="react-tabs"
      data-rttabs="true"
    >
      <ul
        class="react-tabs__tab-list"
        role="tablist"
      >
        <li
          aria-controls="panel:r0:0"
          aria-disabled="false"
          aria-selected="true"
          class="react-tabs__tab react-tabs__tab--selected"
          data-rttab="true"
          id="tab:r0:0"
          role="tab"
          tabindex="0"
        >
          Leaderboard
        </li>
        <li
          aria-controls="panel:r0:1"
          aria-disabled="false"
          aria-selected="false"
          class="react-tabs__tab"
          data-rttab="true"
          id="tab:r0:1"
          role="tab"
        >
          Graphs
        </li>
      </ul>
      <div
        aria-labelledby="tab:r0:0"
        class="react-tabs__tab-panel react-tabs__tab-panel--selected"
        id="panel:r0:0"
        role="tabpanel"
      >
        <h2>
          Leaderboard
        </h2>
        <div
          class="filter-container"
        >
          <span>
            Order by: 
          </span>
          <div
            class="dropdown"
          >
            <div
              class="dropdown label"
            >
              <div
                class="dropdown text"
              >
                Commits
              </div>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium dropdown icon css-i4bv87-MuiSvgIcon-root"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                style="transform: rotate(-90deg);"
                viewBox="0 0 24 24"
              >
                <path
                  d="m7 10 5 5 5-5z"
                />
              </svg>
            </div>
            <div
              class="dropdown container"
              style="display: none; position: absolute;"
            >
              <div
                class="dropdown item"
              >
                Merge Requests
              </div>
              <div
                class="dropdown item"
              >
                Issues
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-labelledby="tab:r0:1"
        class="react-tabs__tab-panel"
        id="panel:r0:1"
        role="tabpanel"
      />
    </div>
  </div>
</DocumentFragment>
`;
